### DISCUSSION TITLE
两个功能上的小建议

### DISCUSSION THREAD

**Discussion by YaeSakuraQvQ** [the1812/Bilibili-Evolved#3881] [功能建议 / Ideas]
1.希望作者大大您可以增加一个**默认关闭b站**“**自动开播**”的功能
之前在[#2085](https://github.com/the1812/Bilibili-Evolved/discussions/2085)说过，您也热心的解决了 但是那个脚本现在貌似因为b站播放器v3改版不起作用了...
2.希望作者大大您可以增加一个**记忆音量大小**的功能
本人的使用场景是chrome无痕+登录账号 但是b站并没有很好的记录这些设置(貌似b站是通过cookies实现的) 
每次重新打开浏览器的时候开视频都是100%音量+自动播放(耳机党直接阵亡好吧) 
(同样，我知道这些功能b站就有开关 可惜不能记住设置 希望可以成为像**记忆倍速**那样的自动化功能)
现在自从Bilibili-Evolved改版到v2后功能都变成组件式的增量安装了，如果大佬有时间而且愿意的话就写一下这个功能吧，非常感谢~
作者大大辛苦~感谢~

**Comment by Kal-tsit63** [comment]
自动开播不是可以在播放器设置里关掉吗？

**Reply by Geeyun-JY3** [reply]
楼主开无痕

**Comment by the1812** [comment]
无痕是不是油猴的设置也记不住

**Reply by YaeSakuraQvQ** [reply]
油猴的设置我可以先调好，然后在无痕里就可以正常用(包括大佬的脚本以及其他chrome扩展也是一样)
只有在无痕里改动设置才不会记住
(然而b站的播放器并不吃我这一套)

**Reply by the1812** [reply]
我看了下可以用啊, 3.x 的播放器设置长得一样的, 我把之前那个粘贴过来加了个音量设置, 就是那个 `volume`, 你可以自己改成别的数值
```js
// ==UserScript==
// @name         禁止自动播放
// @version      1.0.0
// @author       Grant Howard
// @license      MIT
// @match        *://www.bilibili.com/*
// @run-at       document-start
// ==/UserScript==

const settings = JSON.parse(localStorage.getItem('bilibili_player_settings'))
settings.video_status = { ...settings.video_status, autoplay: false, autoplay_has_set: true, autoplay_reddot_status: false, volume: 0.6 }
localStorage.setItem('bilibili_player_settings', JSON.stringify(settings))
```

**Reply by YaeSakuraQvQ** [reply]
~~谢谢大佬，我再试试吧...(主要是之前好多次没有生效，打开视频就自动播放，我才怀疑是b站播放器改版导致的)~~
谢谢哈~
我重新装了这个脚本试了一下，重开浏览器后还是会自动播放+100%音量，请问在您的电脑上脚本是能正常工作的嘛
(我是chrome+油猴 均最新版)
(切换ua后 返回旧版播放器 脚本就又能正常工作...)

**Reply by the1812** [reply]
正常的啊, 我现在一直用的 3.x 播放器

**Reply by the1812** [reply]
我又发现存储里有另一套设置, 看名字应该是新版播放器会使用到的, 试试看?
```js
// ==UserScript==
// @name         禁止自动播放 (for 3.x)
// @version      1.0.0
// @author       Grant Howard
// @license      MIT
// @match        *://www.bilibili.com/*
// @run-at       document-start
// ==/UserScript==

const settings = JSON.parse(localStorage.getItem('bpx_player_profile'))
settings.media = { ...settings.media, autoplay: false, volume: 0.6 }
localStorage.setItem('bpx_player_profile', JSON.stringify(settings))
```

**Reply by YaeSakuraQvQ** [reply]
感谢大佬，完美解决~
